test/roby_app/test_plugin.rb

```diff
@@ -218,20 +218,20 @@ def perform_app_assertion(result)
                     @server_threads = []
                     @process_servers = []
                     register_process_server("test_ps")
-                    Plugin.log_transfer_ip = "127.0.0.1"
-                    Plugin.setup_local_log_transfer_server(Roby.app)
+                    app.log_transfer_ip = "127.0.0.1"
+                    app.setup_local_log_transfer_server
                 end
 
                 after do
                     close_process_servers
-                    Plugin.log_transfer_ip = nil
+                    app.log_transfer_ip = nil
                 end
 
                 it "uploads file from Process Server" do
                     @ps_logfile = create_test_file(@ps_log_dir)
                     path = File.join(app.log_dir, "logfile.log")
                     refute File.exist?(path)
-                    @client = Plugin.send_file_transfer_command("test_ps", @ps_logfile)
+                    @client = app.send_file_transfer_command("test_ps", @ps_logfile)
```

Comment:
What is this @client here ?

Reply:
send_file_transfer_command returns the client instance used to upload  the log so that we can check client.log_upload_state in wait_for_upload_completion

Reply:
Then, same comment than before. Do not pass arguments through instance variables. Give it as argument. I understand it's a side-effect of copy-pasting from the remote processes tests, but the situation there was different (the client was created far away in the setup blocks, in this case it is created and used right away)